### ISSUE TITLE
[accessibility][Android] Unable to provide an accessibility label for a TextField

### ISSUE THREAD

**Issue by math1man** [flutter/flutter#184012]
### Steps to reproduce

1. Run the sample code on Android
2. Turn on TalkBack accessibility
3. Interact with the TextField

### Expected results

TextField accessibility label should include the label "Room height"

### Actual results

TextField accessibility label is "3, Edit box" without the label.

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(useMaterial3: false),
      title: 'Flutter Demo',
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title, style: TextStyle(fontFamily: 'ProductSans')),
      ),
      body: Center(
        child: MergeSemantics(
          child: Semantics(
            label: 'Room Height',
            child: TextField(
              keyboardType: TextInputType.number,
              decoration: const InputDecoration(
                prefixText: 'height',
                suffixText: 'feet',
              ),
            ),
          ),
        ),
      ),
    );
  }
}
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

https://github.com/user-attachments/assets/848480bd-1f30-4831-97f9-4665ffb920c4

</details>

### Logs

<summary>Logs</summary>

N/A, google internal client

### Flutter Doctor output

<summary>Doctor output</summary>

N/A, google internal client

**Comment by hannah-hyj** [comment]
This is actually the similar  cause as https://github.com/flutter/flutter/issues/185266, 

see my comment  in [#185266](https://github.com/flutter/flutter/issues/185266#issuecomment-4616179951).

> This happens because flutter uses `setHintText` in ANI to set the floating label text. 
> 
> And Android assumes that hint text is *only visible* when there's no value. 
> so Android only announces the `hinttext` when value is null, when value is not null,  android omit the hint text announcement. 
> 
> I tested on Android and iOS, when label text and value are both visible. android textfield is actually 2 a11y nodes, iOS textfiels is only  a11y node.
> 
> I think the solution on the flutter requires changes in the framework.
> 
> 1. we should produce two semantics node for android and one semantics node for iOS.  
> 2. and support ANI. labeledBy by adding a semantics property. https://developer.android.com/reference/android/view/accessibility/AccessibilityNodeInfo#addLabeledBy(android.view.View)
>
